src/models.c

```diff
@@ -2650,25 +2650,47 @@ BoundingBox GetMeshBoundingBox(Mesh mesh)
 // Compute mesh tangents
 // NOTE: To calculate mesh tangents and binormals we need mesh vertex positions and texture coordinates
 // Implementation base don: https://answers.unity.com/questions/7789/calculating-tangents-vector4.html
-void GenMeshTangents(Mesh *mesh)
+void GenMeshTangents(Mesh* mesh)
 {
-    if (mesh->tangents == NULL) mesh->tangents = (float *)RL_MALLOC(mesh->vertexCount*4*sizeof(float));
-    else TRACELOG(LOG_WARNING, "MESH: Tangents data already available, re-writting");
+    /*
+     * Added - 7/16/2021 - Jason Penick
+     * Check and see if the mesh already has tangents.
+     * If the mesh does not have tangents we can simply alloate the space for the data.
+     * 
+     * If the mesh did have tangents, we can not readily depend on the current vertexCount
+     * to be the actual length of the existing tangent data as the mesh vertices could
+     * have been altered and now contain less or more vertices which would result in
+     * either updating to few tangents, or attempting to over-run the space allocated
+     * for the tangents previously. In order to account for this the memory for the
+     * current tangent data must be freed as well as the previously assigned VertexBuffer,
+     * and then we can recreate the tangent data with the new size designated by vertexCount
+     * and create a new VertexBuffer to assign for the tangents.
+     * 
+     */
+    if (mesh->tangents == NULL)
+    {
+        mesh->tangents = (float*)RL_MALLOC(mesh->vertexCount * 4 * sizeof(float));
+    }
+    else {
+        rlUnloadVertexBuffer(mesh->vboId[SHADER_LOC_VERTEX_TANGENT]);
+        RL_FREE(mesh->tangents);
+        mesh->tangents = (float*)RL_MALLOC(mesh->vertexCount * 4 * sizeof(float));
+    }
 
-    Vector3 *tan1 = (Vector3 *)RL_MALLOC(mesh->vertexCount*sizeof(Vector3));
-    Vector3 *tan2 = (Vector3 *)RL_MALLOC(mesh->vertexCount*sizeof(Vector3));
+    Vector3* tan1 = (Vector3*)RL_MALLOC(mesh->vertexCount * sizeof(Vector3));
+    Vector3* tan2 = (Vector3*)RL_MALLOC(mesh->vertexCount * sizeof(Vector3));
 
     for (int i = 0; i < mesh->vertexCount; i += 3)
     {
         // Get triangle vertices
-        Vector3 v1 = { mesh->vertices[(i + 0)*3 + 0], mesh->vertices[(i + 0)*3 + 1], mesh->vertices[(i + 0)*3 + 2] };
-        Vector3 v2 = { mesh->vertices[(i + 1)*3 + 0], mesh->vertices[(i + 1)*3 + 1], mesh->vertices[(i + 1)*3 + 2] };
-        Vector3 v3 = { mesh->vertices[(i + 2)*3 + 0], mesh->vertices[(i + 2)*3 + 1], mesh->vertices[(i + 2)*3 + 2] };
+        Vector3 v1 = { mesh->vertices[(i + 0) * 3 + 0], mesh->vertices[(i + 0) * 3 + 1], mesh->vertices[(i + 0) * 3 + 2] };
```

Comment:
Not following [raylib code style](https://github.com/raysan5/raylib/blob/master/CONVENTIONS.md).